web.js

```diff
@@ -24,18 +25,17 @@ module.exports = function web (options) {
   var seneca = this
   var extend = seneca.util.deepextend
 
-  // Avoid deepextending context and auth,
-  // they aren't stringify friendly
-  opts = extend(opts, _.omit(options, ['context', 'auth']))
+  // Avoid deepextending context, middleware and auth, they aren't stringify friendly
+  opts = extend(opts, _.omit(options, ['context', 'auth', 'middleware']))
```

Comment:
remove `middleware` from here.  extending functions should be fine.  To be honest, I'm not sure why `auth` is here, I think that might be a HAPI thing that requires a complicated object....  Context *definitely* shouldn't be serializing as this could be the entire web framework, and would be is slow (also totally blows up with circular references in koa1/koa2).